src/ParameterParser.h

```diff
@@ -196,176 +234,108 @@ namespace snowcrash {
             return nested;
         }
 
-        static void parseSignature(const mdp::MarkdownNodeIterator& node,
-                                   SectionParserData& pd,
-                                   mdp::ByteBuffer& signature,
-                                   const ParseResultRef<Parameter>& out) {
+        template<typename T>
+        static void parseAttributes(const mdp::MarkdownNodeIterator& node,
+                                    SectionParserData& pd,
+                                    const std::vector<mdp::ByteBuffer>& attributes,
+                                    const ParseResultRef<T>& out,
+                                    const bool oldSyntax = true) {
 
             out.node.use = UndefinedParameterUse;
-            TrimString(signature);
-
-            if (pd.sectionContext() == ParameterSectionType) {
+            size_t max = oldSyntax ? OLD_SYNTAX_MAX_ATTRIBUTES : NEW_SYNTAX_MAX_ATTRIBUTES;
 
-                mdp::ByteBuffer innerSignature = signature;
-                innerSignature = TrimString(innerSignature);
+            if (attributes.size() <= max) {
+                size_t i = 0;
+                bool definedUse = false;
 
-                size_t firstSpace = innerSignature.find(" ");
+                // Traverse over parameter's traits
+                while (i < attributes.size()) {
```

Comment:
What about using a `for` loop since there is an increment below?

```diff
- while (i < attributes.size()) {
+ for (; i < attributes.size(); ++i) {
...
- i++;
```